sdk/handler.go

```diff
@@ -38,47 +38,45 @@ func (h Handler) Serve(l net.Listener) error {
 }
 
 // ServeTCP makes the handler to listen for request in a given TCP address.
-// It also writes the spec file on the right directory for docker to read.
+// It also writes the spec file in the right directory for docker to read.
 func (h Handler) ServeTCP(pluginName, addr string, tlsConfig *tls.Config) error {
-	return h.listenAndServe("tcp", addr, pluginName, tlsConfig)
+	return h.listenAndServe(newTCPListener(addr, pluginName, tlsConfig))
 }
 
 // ServeUnix makes the handler to listen for requests in a unix socket.
-// It also creates the socket file on the right directory for docker to read.
+// It also creates the socket file in the right directory for docker to read.
 func (h Handler) ServeUnix(systemGroup, addr string) error {
-	return h.listenAndServe("unix", addr, systemGroup, nil)
+	return h.listenAndServe(newUnixListener(addr, systemGroup))
+}
+
+// ServeWindows makes the handler to listen for request in a windows named pipe.
+// It also creates the spec file in the right directory for docker to read.
+func (h Handler) ServeWindows(addr, pluginName string, pipeConfig *WindowsPipeConfig) error {
+	return h.listenAndServe(newWindowsListener(addr, pluginName, pipeConfig))
 }
 
 // HandleFunc registers a function to handle a request path with.
 func (h Handler) HandleFunc(path string, fn func(w http.ResponseWriter, r *http.Request)) {
 	h.mux.HandleFunc(path, fn)
 }
 
-func (h Handler) listenAndServe(proto, addr, group string, tlsConfig *tls.Config) error {
-	var (
-		err  error
-		spec string
-		l    net.Listener
-	)
+type newListenerFunc func() (net.Listener, string, string, error)
+
+func (h Handler) listenAndServe(fn newListenerFunc) error {
+
+	listener, addr, spec, err := fn()
```

Comment:
Hey, thanks for review. 

I actually thought about the same thing when I was writing this, but I really don't like current signature of listenAndServe.

Notice how currently, "group" argument is used both as system group (in case of unix sockets) and plugin name (in case of tcp). It seems like unix sockets were implemented first, and when TCP was added, someone just reused "group" parameter just because it's a string. I really think that listenAndServe function should have been refactored right then.